blaze-server/src/main/scala/org/http4s/blaze/server/BlazeServerBuilder.scala

```diff
@@ -209,6 +209,18 @@ class BlazeServerBuilder[F[_]] private (
   override def bindSocketAddress(socketAddress: InetSocketAddress): Self =
     copy(socketAddress = socketAddress)
 
+
+  /** Configures the compute thread pool used to process requests
+    *
+    * This defaults to `cats.effect.Async[F].executionContext`. In
+    * almost all cases, it is desirable to use the default.
+    *
+    * The Blaze server has a single-threaded event loop receiver used
+    * for picking up tcp connections which is completely separate to
+    * this pool. Following picking up a tcp connection, Blaze shifts
+    * to this compute pool to process requests. Request processing
+    * logic is specified by the `HttpApp`.
```

Comment:
> Request processing logic is specified by the `HttpApp`
On re-reading, I don't think this is correct. The handlers specified by `HttpApp[IO]` would run on the `Async[F].executionContext` pool instead, wouldn't they. It would just be Futures defined internally to blaze that would run on this custom execution context.

Reply:
I understood this deeply a few effect systems ago, but had to do some experiments now!  Using `withExecutionContext(scala.concurrent.ExecutionContext.global)`, with this route:

```
      case GET -> Root / "thread" =>
        F.delay(Thread.currentThread.getName).flatMap(Ok(_))
```

* The response body is `io-compute-#`.  I'd say this is the "logic specified by the `HttpApp`".
* `reqLoopCallback` gets called on `blaze-selector-#` 
* `runRequest` gets called on `io-compute-#`
* `renderResponse` gets called on `io-compute-#`.
* The effect of writing the body with bodyEncoder gets called on `io-compute-#`.  req loop callback gets called again on the same thread.

So what actually does happen on the ExecutionContext?

* The runnable after we have parsed a request, but we were already on a compute pool.
* Some scheduled tasks, like the idle timeout
* Probably some onComplete handlers in blaze that I can't trace from this repo
* Every `EntityBodyWriter` has one, but I compiled without it!

It's a lot less useful than it used to be!  I think the less-detailed description is better for now, and we might be able to rip out more places it's used with the advances in CE3.